tasks/main.yml

```
@@ -89,19 +89,27 @@
   import_tasks: virtualenv.yml
   when: ansible_os_family in ['RedHat', 'Darwin']
 
+- name: define private key path
+  stat:
+    path: "{{ ler53_cert_dir }}/{{ ler53_key_file_name }}"
+  register: private_key_path
+  tags:
+  - openssl
+
 - name: generate the private key
   openssl_privatekey:
-    path: "{{ ler53_cert_dir }}/{{ ler53_key_file_name }}"
+    path: "{{ private_key_path.stat.path }}"
     size: "{{ ler53_key_size }}"
     owner: "{{ ler53_cert_files_owner }}"
     group: "{{ ler53_cert_files_group }}"
     mode: "{{ ler53_cert_files_mode }}"
+  when: not private_key_path.stat.exists or not ler53_static_private_key
```

I just want either the key to be managed by Ansible or not. This behavior is too complex.
```suggestion
  when: not ler53_static_private_key
```

---

Implemented as you've suggested.